using-valtimo/form-flow/create-form-flow-definition.md

```diff
@@ -17,119 +17,157 @@ To create form flow definition, the following steps are necessary:
 
 * Add each individual step to the form flow definition, e.g.:
 
-    ```json
-    {
-        "startStep": "start",
-        "steps": [
-            {
-                "key": "start"
-            },
-            {
-                "key": "step2"
-            },
-            {
-                "key": "step3"
-            },
-            {
-                "key": "end"
-            }
-        ]
-    }
-    ```
+```json
+{
+    "startStep": "personalDetailsStep",
+    "steps": [
+        {
+            "key": "personalDetailsStep"
+        },
+        {
+            "key": "loanApprovedStep"
+        },
+        {
+            "key": "loanDeniedStep"
+        },
+        {
+            "key": "summaryStep"
+        }
+    ]
+}
+```
 
 * Add the next steps for each of the individual steps. This can be done through the `nextStep` property, which
 supports a single step, or `nextSteps`, which supports multiple next steps. Each step should correspond to the key
-of another step defined in this form flow. Since expressions are not yet supported for `nextSteps`, the first step
-will always be selected. So in this example, step 2 will always be followed by step 3.
-
-    ```json
-    {
-        "startStep": "start",
-        "steps": [
-            {
-                "key": "start",
-                "nextStep": "step2"
-            },
-            {
-                "key": "step2",
-                "nextSteps": [
-                    {
-                        "step": "step3"
-                    },
-                    {
-                        "step": "end"
-                    }
-               ]
-            },
-            {
-                "key": "step3",
-                "nextStep": "end"
-            },
-            {
-                "key": "end"
-            }
-        ]
-    }
-    ```
+of another step defined in this form flow.
+
+```json
+{
+    "startStep": "personalDetailsStep",
+    "steps": [
+        {
+            "key": "personalDetailsStep",
+            "nextSteps": [
+                {
+                    "step": "loanApprovedStep"
+                },
+                {
+                    "step": "loanDeniedStep"
+                }
+           ]
+        },
+        {
+            "key": "loanApprovedStep",
+            "nextStep": "summaryStep"
+        },
+        {
+            "key": "loanDeniedStep",
+            "nextStep": "summaryStep"
+        },
+        {
+            "key": "summaryStep"
+        }
+    ]
+}
+```
+
+* The `personalDetailsStep` is now followed by two other steps. This is only allowed when at least one of the two steps
```

Comment:
If it is required to have at least one conditional step, please don't given an example (above) that is invalid.

Reply:
The first few examples also miss a `type` which makes them invalid. It section just gives a step by step approach to build a valid form flow

Reply:
I think it's okay to keep to this approach. This page is meant to be followed step by step, so it makes sense to only introduce the elements you're talking about at that point in time.